src/main/java/org/wso2/carbon/inbound/custom/poll/MQPollingInbound.java

```diff
@@ -0,0 +1,221 @@
+            Builder builder;
+            if (StringUtils.isEmpty(contentType)) {
+                logger.warn("Unable to determine content type for message, setting to text/plain for " + name);
+                contentType = MQConstant.DEFAULT_CONTENT_TYPE;
+            }
+            int index = contentType.indexOf(';');
+            String type = index > 0 ? contentType.substring(0, index) : contentType;
+            builder = BuilderUtil.getBuilderFromSelector(type, axis2MsgCtx);
+            if (builder == null) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("No message builder found for type '" + type +
+                            "'. Falling back to SOAP. for" + name);
+                }
+                builder = new SOAPBuilder();
+            }
+            OMElement documentElement = builder.processDocument(in, contentType, axis2MsgCtx);
+            msgCtx.setEnvelope(TransportUtils.createSOAPEnvelope(documentElement));
+            if (this.injectingSeq == null || "".equals(this.injectingSeq)) {
+                logger.error("Sequence name not specified. Sequence : " + this.injectingSeq + " for " + name);
+                return false;
+            }
+            SequenceMediator seq = (SequenceMediator) this.synapseEnvironment.getSynapseConfiguration()
+                    .getSequence(this.injectingSeq);
+            seq.setErrorHandler(this.onErrorSeq);
+            if (seq != null) {
+                if (logger.isDebugEnabled()) {
+                    logger.debug("injecting message to sequence : " + this.injectingSeq + " of " + name);
+                }
+                if (!this.synapseEnvironment.injectInbound(msgCtx, seq, this.sequential)) {
+                    return false;
+                }
+            } else {
+                logger.error("Sequence: " + this.injectingSeq + " not found for " + name);
+            }
+
+        } catch (Exception e) {
+            throw new SynapseException("Error while processing the IBM MQ Message ", e);
+        }
+        return true;
+    }
+
+    /**
+     * Create the message context.
+     */
+    private MessageContext createMessageContext() {
+        MessageContext msgCtx = this.synapseEnvironment.createMessageContext();
+        org.apache.axis2.context.MessageContext axis2MsgCtx = ((Axis2MessageContext) msgCtx).getAxis2MessageContext();
+        axis2MsgCtx.setServerSide(true);
+        axis2MsgCtx.setMessageID(UUID.randomUUID().toString());
+        return msgCtx;
+    }
+
+    /**
+     * Stopping the inbound endpoint
+     */
+    public void destroy() {
+
+        logger.info("Removing resources");
+        try {
+            if (manager.isConnected()) {
+                manager.close();
+                manager = null;
+            } else {
+                manager = null;
+            }
+        } catch (MQException e) {
+            logger.error("Error removing services");
```

Comment:
Exception is dropped from the log error. Please include exception in log.
logger.error("Error removing services", e);